controllers/usersignup/usersignup_controller_test.go

```diff
@@ -4837,20 +4931,124 @@ func TestUserReactivatingWhileOldSpaceExists(t *testing.T) {
 	})
 }
 
-func (r *Reconciler) setSpaceToReady(name string) error {
+func TestRecordProvisionTime(t *testing.T) {
+	t.Run("should be in histogram", func(t *testing.T) {
+		// given
+		t.Cleanup(metrics.Reset)
+		for i := 0; i < 100; i++ {
+			userSignup := commonsignup.NewUserSignup(
+				commonsignup.WithRequestReceivedTimeAnnotation(time.Now().Add(-time.Duration(i) * time.Second)))
+			client := test.NewFakeClient(t, userSignup)
+
+			// when
+			err := recordProvisionTime(context.TODO(), client, userSignup)
+
+			// then
+			require.NoError(t, err)
+			AssertThatUserSignup(t, test.HostOperatorNs, userSignup.Name, client).DoesNotHaveAnnotation(toolchainv1alpha1.UserSignupRequestReceivedTimeAnnotationKey)
+		}
+		for _, value := range []int{1, 2, 3, 5, 8, 13, 21, 34, 55, 89} {
```

Comment:
very minor - I was wondering where the buckets are coming from, If I'm not mistaken those are defined and created in the `metrics` package, should we extract this to a variable in there ?

Reply:
the buckets were defined by me :-) I decided to use the fibonacci sequence.
As for extracting the buckets into an exported variable. Yeah, we could do that, but assuming some sequence in the unit-tests is a valid approach too, so it starts failing (as an reminder) as soon as we decide to change the buckets in the metrics.

Reply:
it would actually complicate the things - the particular cases use different types (`float64` vs `int` vs `uint64`), so I would keep it as it is

Reply:
Thanks for looking into it !